src/components/accordion/accordion-group/accordion-group.component.tsx

```diff
@@ -0,0 +1,112 @@
+import React, { useMemo, useCallback } from "react";
+import { MarginProps } from "styled-system";
+import invariant from "invariant";
+
+import Events from "../../../__internal__/utils/helpers/events";
+import Accordion, {
+  AccordionInternalProps,
+  AccordionProps,
+} from "../accordion.component";
+import { StyledAccordionGroup } from "../accordion.style";
+
+type AccordionGroupChild =
+  | React.ReactElement
+  | boolean
+  | null
+  | undefined
+  | AccordionGroupChildArray;
+// typescript-to-proptypes breaks on recursive type references so it has to be an interface
+// eslint-disable-next-line @typescript-eslint/no-empty-interface
+interface AccordionGroupChildArray extends Array<AccordionGroupChild> {}
+
+export interface AccordionGroupProps extends MarginProps {
+  children?: AccordionGroupChild;
+}
+
+export const AccordionGroup = ({ children, ...rest }: AccordionGroupProps) => {
+  const hasProperChildren = useMemo(() => {
+    let hasAccordionChildren = true;
+
+    React.Children.toArray(children).forEach((child) => {
+      if (
+        typeof child === "string" ||
+        (React.isValidElement(child) && child.type !== Accordion)
```

Comment:
Apologies for raising this now rather than when I reviewed before. I think we need to use `displayName` here so that any consuming projects who have wrapped the `Accordion` will be able to continue to do so by setting the `displayName` on the wrapping component otherwise this is a breaking change.

```
const Foo = (props) => (<div><Accordion {...props} title="Heading" /></div>
Foo.displayName="Accordion";

Reply:
good point, changed that back to displayName